plugins/woocommerce/tests/e2e-pw/tests/merchant/customer-list.spec.js

```diff
@@ -116,10 +116,16 @@ test.describe( 'Merchant > Customer List', { tag: '@services' }, () => {
 			let x = 1;
 			for ( const customer of customers ) {
 				await page
-					.locator( '#woocommerce-select-control-0__control-input' )
+					.getByRole( 'combobox', {
+						expanded: false,
+						disabled: false,
+					} )
```

Comment:
These tests were targeting the select control input by `id` which included a generated `index`, this seemed like a bad idea, especially if React 18 supports concurrent mode, meaning these indexes could depend on timing.

Reply:
I only figured it out by looking at the video of the failed test and comparing it to my local version. Noticing it was adding the email to the wrong select control in the failing test.